node.js/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -1,12 +1,61 @@
+const { request, response } = require("express");
 const express = require("express");
 const router = express.Router();
 
 const meals = require("./../data/meals.json");
 
 router.get("/", async (request, response) => {
   try {
-    console.log(meals);
-    console.log("in /api/meals");
+    const maxPrice = request.query.maxPrice
+    const title = request.query.title
+    const createdAfter = request.query.createdAfter
+    const limit = request.query.limit
+    if (maxPrice) {
+      if (!Number.isInteger(parseInt(maxPrice))) {
+        response.statusCode = 400;
+        response.json({ "message": "bad input" });
+        return;
+      } else {
+        response.send(meals.filter(meal => meal.price <= maxPrice))
+      }
+    } else if (title) {
+      response.send(meals.filter(meal => meal.title.includes(title)))
+    } else if (createdAfter) {
+      if (!Date.parse(createdAfter)) {
+        response.statusCode = 400;
+        response.json({ "message": "bad input" });
+        return;
+      } else {
+        response.send(meals.filter(meal => Date.parse(meal.createdAt) >= Date.parse(createdAfter)))
+      }
+    } else if (limit) {
+      if (!Number.isInteger(parseInt(limit))) {
+        response.statusCode = 400;
+        response.json({ "message": "bad input" });
+        return;
+      } else {
+        response.send(meals.slice(0, limit))
+      }
+    } else {
+      response.send(meals);
+    }
+  } catch (error) {
+    throw error
+  }
```

Comment:
Overall good with request queries, but what if there are multiple queries at a time?
You need to assign meals to a temp value and that temp value should be passed through all the if loops of queries.
for ex: 
```
tempMeals=meals;

tempMeals=meals.filter(meal => meal.title.includes(title));
response.send(tempMeals);

tempMeals=meals.filter(meal => meal.title.includes(title));
respond.send(tempMeals);

```
and so on..